src/memary/agent/llm_api/tools.py

```diff
@@ -52,3 +49,31 @@ def openai_chat_completions_request(url, api_key, data):
         # Handle other potential errors
         logging.warning(f"Got unknown Exception, exception={e}")
         raise e
+
+
+def ollama_chat_completions_request(messages, model):
+    """sends chat request to model running on Ollama"""
+
+    url = "http://localhost:11434/api/chat"
+    data = {"model": "llama3", "messages": messages, "stream": False}
```

Comment:
And this was the reason why the llama model couldn't recognize our initial persona.
In the [`def _init_persona_to_message(self)`](https://github.com/kingjulio8238/memary/blob/e08cd78eccbaaee57465dda1291e8c2234844f72/src/memary/agent/data_types.py#L55) of our Message types data class, we initialize our persona with "system" and "user" system instruction for ChatGPT API.
But I found [ollama's API](https://github.com/ollama/ollama/blob/main/docs/api.md#request-json-mode) inputs don't provide those system message inputs to leverage persona information.
As for my search results, I think there is no way to integrate the persona info directly into the Ollama API. Let's set persona inputs like this.
```
{
  "model": "llama3",
  "prompt": "system persona: This is your 'system persona', which you should follow to make answers by this instruction. The next message's 'user persona some more explanations~~",
  "format": "json",
  "stream": false
},
{
  "model": "llama3",
  "prompt": "user persona: My name is Mr. Han. I have worked as an ML Engineer for three years.",
  "format": "json",
  "stream": false
}
```

And these persona messages should be the first and second messages.
I hope this trick will work.

Reply:
Seems like the current implementation uses the api/chat endpoint which takes into account the series of messages just like the ChatGPT completion does. When printing the `messages` list in `openai_chat_completions_request`, it is including the system and user persona as the first two messages already, so it seems to be being processed.

Reply:
Yes, you are right in the `openai_chat_completions_request` case for the persona taking.
```
  'model': 'gpt-3.5-turbo',
  'messages': [
  {
    'role': 'system',
    'content': "..."
  },
  {
  'role': 'user',
  'content': "..."
  }, 
```
But in the Ollama, they don't provide a `role` key that we can inject personas into the message.
```
{
  "model": "llama3",
  "prompt": "...",
  "format": "json",
  "stream": false
},
```
So, as I mentioned in the above comment, we should change our `role: system` message format with `"prompt": "system persona: ..." like this format.

Reply:
Sure yep I can try that. Although is this section of the api docs you're referencing?![image](https://github.com/kingjulio8238/memary/assets/66340176/dee4ff3e-a1db-43be-8e1e-74f50917f247) Seems like you're able to add a role if you do chat completions, or am I misinterpreting it?

Reply:
I'll go ahead and try your suggestion though

Reply:
Ohhhh, I just saw that below part. We can change this, `data = {"model": "llama3", "messages": messages, "stream": False}` to like this.
```
{
  "model": "llama3",
  "messages": [
    {
      "role": "system",
      "content": "[system persona] ...."
    }
  ]
}
```

Reply:
Ok sweet! Since `messages` already includes the two different personas and their associated roles (from `_init_persona_to_messages` in `data_types.py`), would any change be necessary?

Reply:
Nothing really.
Just changing [this format](https://github.com/kingjulio8238/memary/blob/e08cd78eccbaaee57465dda1291e8c2234844f72/src/memary/agent/llm_api/tools.py#L58) is all we need!